src/main/kotlin/com/exactpro/th2/check1/utils/TimeUtils.kt

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2021-2021 Exactpro (Exactpro Systems Limited)
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.check1.utils
+
+import com.google.protobuf.Timestamp
+
+private val timestampComparator = Comparator.comparingLong(Timestamp::getSeconds).thenComparingInt(Timestamp::getNanos)
+operator fun Timestamp.compareTo(other: Timestamp): Int = timestampComparator.compare(this, other)
+
+fun Timestamp.isBefore(other: Timestamp): Boolean = this < other
+fun Timestamp.isAfter(other: Timestamp): Boolean = this > other
```

Comment:
Maybe add the isAfterOrEqual method.
`return checkpointTimeout == null || checkpointTimeout!!.isAfter(timestamp) || checkpointTimeout == timestamp`